analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -178,6 +179,29 @@ class GoMod(
         return result
     }
 
+    /**
```

Comment:
Instead of the functions `filterUsedPackages()` and `parseWhyOutput()`, I'd prefer
`getUsedPackages(): Set<Identifier>`. This IMO better aligns with Single responsibiltiy principle and minimized the dependencies of the functions, e.g. the function which determines the used packages does not involve a `Graph` and thus is easier to test / can be tested independently of a Graph.

The implementation then could simply do `graph.subgraph(getUsedPackages())`.

Reply:
Makes sense. However, the `parseWhyOutput()` function has been exposed, so that it can be tested separately, also including corner cases of the output of the `why` command. With the two functions being merged, this would not be possible any more in an easy way. Any suggestions how to achieve this?

Reply:
Sorry, I wrote it wrongly. I didn't mean to ask to ditch `parseWhyOutput()`. Having that for the tests makes sense to me. Replacing `filterUsedPackages()` with a combination of `getUsedPackages` and `graph.subgraph(getUsedPackages())` was more what I was after.